router/core/graphql_prehandler.go

```diff
@@ -217,7 +217,7 @@ func (h *PreHandler) Handler(next http.Handler) http.Handler {
 
 		executionOptions, traceOptions, err := h.parseRequestOptions(r, clientInfo, requestLogger)
 		if err != nil {
-			requestContext.error = err
+			AssignErrors(requestContext, err)
```

Comment:
Instead of writing a utility function to assign the error, you could implement `ResolveAnyExpression` similar as with`ResolveStringExpression` on the `requestContext`. This would follow the pattern and the error can be set right before we pass the expressionContext to the template engine.

Reply:
Agree on the `ResolveAnyExpression`, however I want to clarify about assigning errors in the `ResolveAnyExpression` function.

My assumption was that whenever an error occurred that the error should be available as part of the expressionContext. So if anyone did 
`reqContext.expressionContext.errror` at any time they wont get null (which would be different from `requestContext.error`).

With this change they would get null, but from what you are saying is that they should not access the expressionContext this way so **its ok**, instead they should access it via any `Resolve*Expression` method, for which we will set the error to the expressionContext before the resolving step?

Reply:
Correct. Thanks for the explanation.